pkg/datastore/phpapi/phpapi.go

```diff
@@ -48,59 +50,76 @@ func (p *PHPAPI) GetClusters(colorCodes []string) ([]types.SubmarinerCluster, er
 	klog.V(8).Infof("request url: %s", requestURL)
 	clustersGetter, err := http.Get(requestURL)
 	if err != nil {
-		klog.Errorf("Encountered error while trying to retrieve clusters from the apiserver: %v", err)
-		return nil, err
+		return nil, fmt.Errorf("error retrieving clusters from %s: %v", requestURL, err)
 	}
+
 	defer clustersGetter.Body.Close()
 	clustersRaw, err := ioutil.ReadAll(clustersGetter.Body)
-	var clusters []types.SubmarinerCluster
-	klog.V(8).Infof("response body: %v", string(clustersRaw[:]))
-	// let's unmarshal into our type the cluster into
-	err = json.Unmarshal(clustersRaw, &clusters) // need to actually make the API return json that works for this
 	if err != nil {
-		klog.Errorf("Error while unmarshaling JSON from clusters")
-		return nil, err
+		return nil, fmt.Errorf("error reading response from %s: %v", requestURL, err)
+	}
+
+	klog.V(8).Infof("Response body: %v", string(clustersRaw[:]))
+
+	// let's unmarshal into our type the cluster into
+	// need to actually make the API return json that works for this
+	var clusters []types.SubmarinerCluster
+	if err = json.Unmarshal(clustersRaw, &clusters); err != nil {
+		return nil, fmt.Errorf("error unmarshalling JSON %s: %v", string(clustersRaw[:]), err)
 	}
 	return clusters, nil
 }
+
 func (p *PHPAPI) GetCluster(clusterID string) (types.SubmarinerCluster, error) {
 	requestURL := fmt.Sprintf("%s://%s/clusters.php?plurality=false&identifier=%s&cluster_id=%s", p.Proto, p.Server, p.APIToken, clusterID)
 	klog.V(8).Infof("request url: %s", requestURL)
 	clustersGetter, err := http.Get(requestURL)
 	if err != nil {
-		klog.Errorf("Encountered error while trying to retrieve clusters from the apiserver")
-		return types.SubmarinerCluster{}, err
+		return types.SubmarinerCluster{}, fmt.Errorf("error retrieving clusters from %s: %v", requestURL, err)
```

Comment:
I know this is what the code already did, but wouldn’t it be better to return `nil` with the error?

Reply:
yeah probably but that changes the interface and call sites and outside the scope of this patch.